components/brave_wallet_ui/common/async/handlers.ts

```diff
@@ -124,6 +124,10 @@ handler.on(WalletActions.keyringRestored.getType(), async (store) => {
   await refreshWalletInfo(store)
 })
 
+handler.on(WalletActions.keyringReset.getType(), async (store) => {
+  window.location.reload()
```

Comment:
should it also make `getAPIProxy().braveWalletService.reset()` call?

Reply:
No the reset is initiated by the settings page, this just listens to when it is reset so it can reload the page to show the create form again. It was easier than trying to reset all of the redux state and works fine.